lnwallet/test_utils.go

```diff
@@ -529,3 +543,209 @@ func ForceStateTransition(chanA, chanB *LightningChannel) error {
 
 	return nil
 }
+
+// GetCommitmentOutChecker returns a function which validates that we're
+// signing an output we can derive given a remote signer's knowledge of the
+// channel from initial negotiation. It can handle the outputs of a commitment
+// or second-level HTLC transaction or a cooperative close of the channel.
+func GetCommitmentOutChecker(t *testing.T, chanType channeldb.ChannelType,
+	initiator bool, local, remote channeldb.ChannelConfig) func(
+	*wire.TxOut, map[string][]byte) error {
+
+	byteOrder := binary.LittleEndian
+
+	return func(out *wire.TxOut, unkMap map[string][]byte) error {
+		// First, check for commitment anchor output.
+		if chanType.HasAnchors() && out.Value == int64(anchorSize) &&
+			len(unkMap) == 0 {
+
+			return nil
+		}
+
+		// Next, get our info from the map for easy use in checking
+		// derivations.
+		var (
+			lease, csv, cltv                     uint32
+			commitPoint                          *btcec.PublicKey
+			keyRingOurCommit, keyRingTheirCommit *CommitmentKeyRing
+			rHash                                [32]byte
+			err                                  error
+		)
+
+		leaseBytes, hasLease :=
+			unkMap[string(input.PsbtKeyTypeOutputLeaseExpiry)]
```

Comment:
nit: create local alias for these string converted keys to make the code a bit more easy to read?